client/db/src/lib.rs

```diff
@@ -477,7 +477,6 @@ impl<Block: BlockT> BlockchainDb<Block> {
 		let mut meta = self.meta.write();
 		if number.is_zero() {
 			meta.genesis_hash = hash;
-			meta.finalized_hash = hash;
```

Comment:
Why is that removed?

Reply:
This is not really needed. Genesis is explicitly finalized in `try_commit_operation` anyway. It is also wrong because we don't want to re-finalize genesis when restarting interrupted fast sync.